x-pack/plugins/stack_alerts/server/alert_types/geo_containment/alert_type.ts

```diff
@@ -0,0 +1,177 @@
+import { schema } from '@kbn/config-schema';
+import { Logger } from 'src/core/server';
+import { STACK_ALERTS_FEATURE_ID } from '../../../common';
+import { getGeoContainmentExecutor } from './geo_containment';
+import {
+  ActionGroup,
+  AlertServices,
+  ActionVariable,
+  AlertTypeState,
+} from '../../../../alerts/server';
+import { Query } from '../../../../../../src/plugins/data/common/query';
+
+export const GEO_CONTAINMENT_ID = '.geo-containment';
+export const ActionGroupId = 'Tracked entity contained';
+
+const actionVariableContextEntityIdLabel = i18n.translate(
+  'xpack.stackAlerts.geoContainment.actionVariableContextEntityIdLabel',
+  {
+    defaultMessage: 'The entity ID of the document that triggered the alert',
+  }
+);
+
+const actionVariableContextEntityDateTimeLabel = i18n.translate(
+  'xpack.stackAlerts.geoContainment.actionVariableContextFromEntityDateTimeLabel',
+  {
+    defaultMessage: `The date the entity was recorded in the boundary`,
+  }
+);
+
+const actionVariableContextEntityDocumentIdLabel = i18n.translate(
+  'xpack.stackAlerts.geoContainment.actionVariableContextFromEntityDocumentIdLabel',
+  {
+    defaultMessage: 'The id of the contained entity document',
+  }
+);
+
+const actionVariableContextDetectionDateTimeLabel = i18n.translate(
+  'xpack.stackAlerts.geoContainment.actionVariableContextDetectionDateTimeLabel',
+  {
+    defaultMessage: 'The alert interval end time this change was recorded',
+  }
+);
+
+const actionVariableContextEntityLocationLabel = i18n.translate(
+  'xpack.stackAlerts.geoContainment.actionVariableContextFromEntityLocationLabel',
+  {
+    defaultMessage: 'The location of the entity',
+  }
+);
+
+const actionVariableContextContainingBoundaryIdLabel = i18n.translate(
+  'xpack.stackAlerts.geoContainment.actionVariableContextContainingBoundaryIdLabel',
+  {
+    defaultMessage: 'The id of the boundary containing the entity',
+  }
+);
+
+const actionVariableContextContainingBoundaryNameLabel = i18n.translate(
+  'xpack.stackAlerts.geoContainment.actionVariableContextContainingBoundaryNameLabel',
+  {
+    defaultMessage: 'The boundary the entity is currently located within',
+  }
+);
+
+const actionVariables = {
+  context: [
+    // Alert-specific data
+    { name: 'entityId', description: actionVariableContextEntityIdLabel },
+    { name: 'entityDateTime', description: actionVariableContextEntityDateTimeLabel },
+    { name: 'entityDocumentId', description: actionVariableContextEntityDocumentIdLabel },
+    { name: 'detectionDateTime', description: actionVariableContextDetectionDateTimeLabel },
+    { name: 'entityLocation', description: actionVariableContextEntityLocationLabel },
+    { name: 'containingBoundaryId', description: actionVariableContextContainingBoundaryIdLabel },
+    {
+      name: 'containingBoundaryName',
+      description: actionVariableContextContainingBoundaryNameLabel,
+    },
+  ],
+};
+
+export const ParamsSchema = schema.object({
```

Comment:
+1 I think this probsbly capture what we would like in a message. cc @kmartastic Anything else?